File: server/readable/web/js/model/emotion_point.js
/*
Copyright 2020 NEC Solution Innovators, Ltd.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/
function EmotionPointList() {
    ArrayList.call(this);
};(function() {
  var Super = function Super() {
  };
  Super.prototype = ArrayList.prototype;
  EmotionPointList.prototype = new Super();
  var _super = Super.prototype;
  var _proto = EmotionPointList.prototype;
  _proto.add = function(emotionPointData) {
      var _self = this;
      if(emotionPointData == null || typeof emotionPointData != 'object') {
          return false;
      }
      if(emotionPointData.getJid == null) {
          return false;
      }
      var _jid = emotionPointData.getJid();
      if(_jid == null || _jid == '') {
          return false;
      }
      var _existEmotionPointData = _self.getByJid(_jid);
      if(_existEmotionPointData != null) {
          return false;
      }
      return _super.add.call(this, emotionPointData);
  };
  _proto.getByJid = function(jid) {
      var _self = this;
      if (jid == null || typeof jid != 'string') {
          return null;
      }
      var _retEmotionPointData = null;
      var _count = _self._length;
      for (var i = 0; i<_count; i++) {
          if (_self._array[i].getJid() == jid) {
              _retEmotionPointData = _self._array[i];
              break;
          }
      }
      return _retEmotionPointData;
  };
  _proto.removeByJid = function(jid) {
      var _self = this;
      if(jid == null || typeof jid != 'string') {
          return null;
      }
      var _retEmotionPointData = null;
      var _count = _self._length;
      for(var _i = 0; _i < _count; _i++) {
          if(_self._array[_i].getJid() == jid) {
              _retEmotionPointData = _super.remove.call(this, _i);
              break;
          }
      }
      return _retEmotionPointData;
  };
  _proto.removeAll = function() {
      _super.removeAll.call(this);
  };
})();

 function EmotionPointData() {
    Profile.call(this);
    this._jid = '';
    this._created_at = '';
    this._updated_at = '';
    this._emotion_point = '';
 }; (function() {
    EmotionPointData.prototype = $.extend({}, Profile.prototype);
    var _proto = EmotionPointData.prototype;

    _proto.getJid = function() {
        return this._jid;
    }
    _proto.setJid = function(jid) {
        if(jid == null || typeof jid != 'string') {
            return;
        }
        if(jid == '') {
            return;
        }
        this._jid = jid;
    }
    _proto.getCreatedAt = function() {
        return this._created_at;
    }
    _proto.setCreatedAt = function(created_at) {
        if(created_at == null || typeof created_at != 'object') {
            return;
        }
        if(created_at == '') {
            return;
        }
        this._created_at = created_at;
    }
    _proto.getUpdatedAt = function() {
        return this._updated_at;
    }
    _proto.setUpdatedAt = function(updated_at) {
        if(updated_at == null || typeof updated_at != 'object') {
            return;
        }
        if(updated_at == '') {
            return;
        }
        this._updated_at = updated_at;
    }
    _proto.getEmotionPoint = function() {
        return this._emotion_point;
    }
    _proto.setEmotionPoint = function(emotion_point) {
        if(emotion_point == null || typeof emotion_point != 'number') {
            return;
        }
        // Variable 'emotion_point' is of type number, but it is compared to an expression of type string.
        //if(emotion_point == '') { 
        if(emotion_point == 0) {
            return;
        }
        this._emotion_point = emotion_point;
    }
 })();

File: src/openfire/src/plugins/globalSNS/src/java/jp/co/nec/necst/spf/globalSNS/Data/EmotionPoint.java
/*
Copyright 2020 NEC Solution Innovators, Ltd.
Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at
    http://www.apache.org/licenses/LICENSE-2.0
Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

package jp.co.nec.necst.spf.globalSNS.Data;

import java.math.BigInteger;
import java.sql.Timestamp;
import java.text.SimpleDateFormat;
import org.dom4j.DocumentHelper;
import org.dom4j.Element;
import org.dom4j.Document;
import org.dom4j.DocumentException;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class EmotionPoint {
    private BigInteger mId;
    private String mItemId;
    private String mMsgOwnJid;
    private String mMsgTo;
    private int mMsgType;
    private String mEntry;
    private int mEmotionPoint;
    private String mJid;
    private Timestamp mCreatedAt;
    private Timestamp mUpdatedAt;
    private String emotionIconJson;

    private static final Logger Log = LoggerFactory
        .getLogger(GoodJob.class);

    public EmotionPoint() {
        mId = BigInteger.ZERO;
        mItemId = "";
        mMsgOwnJid = "";
        mMsgTo = "";
        mMsgType = -1;
        mEntry = "";
        mEmotionPoint = 0;
        mJid = "";
        mCreatedAt = null;
        mUpdatedAt = null;
        emotionIconJson = "";
    }

    public EmotionPoint(EmotionPoint src) {
        mId = src.getId();
        mItemId = src.getItemId();
        mMsgOwnJid = src.getMsgOwnJid();
        mMsgTo = src.getMsgTo();
        mMsgType = src.getMsgType();
        mEntry = src.getEntry();
        mEmotionPoint = src.getEmotionPoint();
        mJid = src.getJid();
        mCreatedAt = src.getCreatedAt();
        mUpdatedAt = src.getUpdatedAt();
    }

    public BigInteger getId() {
        return mId;
    }

    public void setId(BigInteger id) {
        mId = id;
    }

    public String getItemId() {
        return mItemId;
    }

    public void setItemId(String itemId) {
        mItemId = itemId;
    }

    public String getMsgOwnJid() {
        return mMsgOwnJid;
    }

    public void setMsgOwnJid(String msgownjid) {
        mMsgOwnJid = msgownjid;
    }

    public void setMsgTo(String msgto) {
        mMsgTo = msgto;
    }

    public String getMsgTo() {
        return mMsgTo;
    }

    public void setMsgType(int msgtype) {
        mMsgType = msgtype;
    }

    public void setMsgType(String msgtype) {
        mMsgType = Integer.parseInt(msgtype);
    }

    public int getMsgType() {
        return mMsgType;
    }

    public void setEntry(String entry) {
        mEntry = entry;
    }

    public String getEntry() {
        return mEntry;
    }

    public String getEntryBody() {
        Document document = null;
        String body = "";
        try {
            document = DocumentHelper.parseText(mEntry);
            body = document.getRootElement().element("body").getText();
        } catch (DocumentException e) {
            Log.error("EmotionPoint.getEntryBody DocumentHelper.parseText error : " + e);
        }
        return body;
    }

    public int getEmotionPoint() {
        return mEmotionPoint;
    }
    public void setEmotionPoint(int emotionPoint) {
        mEmotionPoint = emotionPoint;
    }

    public String getJid() {
        return mJid;
    }

    public void setJid(String jid) {
        mJid = jid;
    }

    public Timestamp getCreatedAt() {
        return mCreatedAt;
    }

    public String getCreatedAtStr() {
        if(mCreatedAt == null) {
            return "";
        }
        long dateTime = mCreatedAt.getTime();
        java.util.Date date = new java.util.Date(dateTime);
        SimpleDateFormat df = new SimpleDateFormat("yyyy/MM/dd HH:mm:ss.SSS");
        return df.format(date);
    }

    public void setCreatedAt(Timestamp date) {
        mCreatedAt = date;
    }

    public Timestamp getUpdatedAt() {
        return mUpdatedAt;
    }

    public String getUpdatedAtStr() {
        if(mUpdatedAt == null) {
            return "";
        }
        long dateTime = mUpdatedAt.getTime();
        java.util.Date date = new java.util.Date(dateTime);
        SimpleDateFormat df = new SimpleDateFormat("yyyy/MM/dd HH:mm:ss.SSS");
        return df.format(date);
    }

    public void setUpdatedAt(Timestamp date) {
        mUpdatedAt = date;
    }
}
